### ISSUE TITLE
tests: 😅 mock consensus height of 0 is recorded in fast forward span

### ISSUE THREAD

**Issue by cratelyn** [penumbra-zone/penumbra#4005]
`block::Builder::execute` does not update the `TestNode<C>` height. it should! so that we don't see logs like this when running tests:

```
  2024-03-12T16:06:19.179075Z TRACE penumbra_mock_consensus::abci: waiting for consensus service
    at crates/test/mock-consensus/src/abci.rs:32
    in penumbra_mock_consensus::abci::end_block
    in penumbra_mock_consensus::block::execute with height: 718, time: 1710259579
    in penumbra_mock_consensus::fast_forward with blocks: 1000, height: 0, fast_forward.blocks: 1000
```

**Comment by cratelyn** [comment]
errant issue. we do update the height, but we record a span field outside of the loop. thus, the `height: 0` is just reflecting where we started. i'm going to close this, it's a very small logging issue, and it will be fixed as part of #3995.

**Comment by cratelyn** [comment]
uncovered another unrelated height value being hard-coded. for the sake of formality, going to cherry-pick changes out of #3995 and mark them as "fixing" this issue.
